File: 1653/1653.py
class Solution:
    def minimum_deletions_to_make_string_balance(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.minimum_deletions_to_make_string_balance(root.left), self.minimum_deletions_to_make_string_balance(root.right))

File: 1653/1653.c
int minimumDeletionsToMakeStringBa(struct TreeNode* root) {
    if (!root) return 0;
    int l = minimumDeletionsToMakeStringBa(root->left);
    int r = minimumDeletionsToMakeStringBa(root->right);
    return 1 + (l > r ? l : r);
}